clients/lsp-clangd.el

```diff
@@ -235,5 +235,26 @@ returned to avoid that the echo area grows uncomfortably."
         (car (s-lines (lsp--render-element contents)))))))
 
 
+(defun lsp-clangd-to-other (&optional arg)
+  "Open the corresponding header/source file or return nil.
+
+If called with ARG universal argument, the file will open in the other window."
+  (interactive "P")
+  (if-let ((other-fname
+            ;; find-workspace might be too brittle if users make their own lsp IDs
+            ;; eg. clangd-remote
+              ;; TODO define an extension with lsp-interace
+              ;; similar to rust-analyzer on lines 278-292 in
+              ;; lsp-protocol.el
```

Comment:
I think you can delete this comment. You need your interfaces only if there are no lsp-mode types already covering this functionality. And lsp-find-workspace is supposed to be used very rarely. In this case, lsp-mode should be able to handle the request directly using lsp-request.